metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -1179,7 +1187,12 @@ def get_tables_for_dataset(
                             sharded_tables[table_name] = table
                         continue
                 else:
-                    table_count = table_count + 1
+                    if (
+                        table.time_partitioning
+                        or "range_partitioning" in table._properties
+                    ):
+                        partitioned_table_count_in_this_batch += 1
+
                     table_items[table.table_id] = table
```

Comment:
Should we be doing this after checking `table_identifier`? I feel like the below if statement is not actually dropping tables, because they've been added to `table_items`. If this is the case, I think it makes sense to put this in an `elif` attached to the `if shard` block